### ISSUE TITLE
Pyright does not detect existing methods in asyncpg.Connection

### ISSUE THREAD

**Issue by HarrySky** [microsoft/pyright#998]
**Describe the bug**
Pyright does not see existing method `asyncpg.Connection.set_type_codec` and gives me this error:
```
Cannot access member "set_type_codec" for type "Connection"
  Member "set_type_codec" is unknownPyright (reportGeneralTypeIssues)
```

Same problem with `execute`, `fetch`, `fetchval`, `transaction`, etc.

**To Reproduce**
Paste code below in VSCode with Pyright, enable `python.analysis.useLibraryCodeForTypes`, `./venv/bin/pip install asyncpg`

**Expected behavior**
Pyright sees that methods exist as Python extension and VS Code in general does (and lets me click `Go To Definition` successfully)

**Screenshots or Code (Minimal, Reproducible Example)**

```python
from typing import Dict, Any, Optional, Union, List
from asyncpg import Connection

def _json_dumps(data: Optional[Union[List[Any], Dict[str, Any]]]) -> Optional[str]:
    pass

def _json_loads(s: Optional[str]) -> Optional[Union[List[Any], Dict[str, Any]]]:
    pass

async def _set_codecs(connection: Connection) -> None:
    """Setup encoding for jsonb"""
    await connection.set_type_codec(
        "jsonb",
        encoder=_json_dumps,
        decoder=_json_loads,
        schema="pg_catalog",
        format="text",
    )
```

**VS Code extension or command-line**
VS Code extension: v1.1.65

**Additional context**
You can change methods in code above and see that Pyright does not see them.

**Comment by erictraut** [comment]
Thanks for reporting this problem. It was also reported earlier today by someone in the [Pyright gitter channel](https://gitter.im/microsoft-pyright/community). It is fixed in version 1.1.66 of Pyright, which I just published.

If you're interested in the details:
> This turns out to be a bug in the asyncpg library that triggered a bug in Pyright's parser. The library uses an invalid expression for the type expression on the addr parameter in the __init__ method in the Connection class (in connection.py). The parameter is annotated as addr: (str, int) or str. This is legal in the Python grammar (and therefore should not generate a syntax error). Pyright's parser was treating it as a syntax error and attempted to do parse recovery by swallowing some tokens. The parse recovery was not effective, and the remainder of the methods defined in the class were treated as being defined outside of the class (at the module level). I've fixed the bug in the Pyright parser.
